src/mapreduce.jl

```diff
@@ -5,7 +5,10 @@
 
 Base.any(A::GPUArray{Bool}) = mapreduce(identity, |, false, A)
 Base.all(A::GPUArray{Bool}) = mapreduce(identity, &, true, A)
-Base.count(pred, A::GPUArray) = Int(mapreduce(pred, +, 0, A))
+
+Base.any(f::Function, A::GPUArray) = mapreduce(f, |, false, A)
```

Comment:
Just noticed that `Base.any` and `Base.all` have defined slightly different semantics from the mapreduce-based implementation here when I was trying to incorporate the changes into #145 :
1. They are short-circuiting evaluations in `Base`. This changes semantics when the predicate has side-effects. Maybe we can just make it very clear in documentation that the GPU implementation is not short-circuiting?
2. `Base` implementations allow `missing` values, and deal with them by following [three-valued logic](https://en.wikipedia.org/wiki/Three-valued_logic). Should we follow the `Base` behaviour here? It might end up with something like:

```julia
Base.any(pred, A::GPUArray) = let t =
    begin
        mapreduce(x -> let v = pred(x)
            ismissing(v) ? (false, true) : (v && true, false) end, # enforce v is `Bool` or `missing`
        ((v1,m1),(v2,m2)) -> (v1 || v2, m1 || m2),
        A; init = (false, false))
    end
    t[1] ? true : (t[2] ? missing : false)
end
```